### PR TITLE
CLN: minor cleaning of Categorical.__init__

### PR DESCRIPTION
**Pull request by topper-123**
At a certain point in the Categorical init, we should for clarity only use ``dtype`` and not intermix ``categories`` or ``ordered``.

This small PR does that.

### CODE REVIEW

File: pandas/core/arrays/categorical.py [hunk 1]
```diff
@@ -376,9 +371,8 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
         null_mask = np.array(False)
 
         # sanitize input
-        if is_categorical_dtype(values):
-            if dtype.categories is None:
-                dtype = CategoricalDtype(values.categories, dtype.ordered)
+        if is_categorical_dtype(values) and dtype.categories is None:
```
Comment (jschendel) [inline pandas/core/arrays/categorical.py]: I think this should remain nested for perf reasons so the `elif` is not hit unnecessarily when the `is_categorical_dtype(values)` condition is met, but `dtype.categories is None` is not met.  I pulled your changes and did some quick timings with/without this being nested, and the difference does look to be significant.

Nested `if` statements (old version):
```python
In [2]: cat_old = pd.Categorical(list('abca')*10**4)

In [3]: cdt_new = pd.api.types.CategoricalDtype(list('bdac'), True)

In [4]: %timeit  pd.Categorical(cat_old, dtype=cdt_new)
799 µs ± 6.92 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Single flat `if` statement (new version):
```python
In [2]: cat_old = pd.Categorical(list('abca')*10**4)

In [3]: cdt_new = pd.api.types.CategoricalDtype(list('bdac'), True)

In [4]: %timeit  pd.Categorical(cat_old, dtype=cdt_new)
2.26 ms ± 12.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

**Review by jschendel — CHANGES_REQUESTED** [review]

**Review by jschendel — APPROVED** [review]
lgtm pending @jreback's asv comment

**Comment by topper-123** [comment]
I have some problem getting asv working. I don’t think this can have a performance effect now, after the change after @jschendel comments...

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**